climpred/toy.py

```diff
@@ -0,0 +1,259 @@
+import matplotlib.pyplot as plt
+import numpy as np
+import xarray as xr
+from climpred.bootstrap import bootstrap_compute
+from climpred.prediction import compute_perfect_model
+from climpred.tutorial import load_dataset
+
+# standard setup
+lead = xr.DataArray(np.arange(0, 20, 1), dims='lead')
+lead['lead'] = lead.values
+
+control = load_dataset('MPI-control-1D')['tos'].isel(period=-1, area=1)
+# amplitudes of the signal and noise
+noise_amplitude = control.std().values * 2.5
+signal_amplitude = control.std().values
+# period of potentially predictable variable
+signal_P = 8
+
+
+def ramp(lead, a=0.2, t_opt=0.1, r=1.8):
+    """A weighting function that starts at 0 and approaches 1 to mimick increasing noise.
+
+    Args:
+        lead (xr.DataArray): lead times to create synthetic data for.
+        a (type): shape of ramping up. Defaults to 0.2.
+        t_opt (type): lead time near to saturation. Defaults to 0.1.
+        r (type): intensity of ramping up. Defaults to 1.8.
+
+    Returns:
+        np.array: weighting ramping up to 1.
+
+    """
+    A = 1.0 / (1 + a * np.exp((t_opt - lead) / r)) - 1.0 / (1 + a * np.exp((t_opt) / r))
```

Comment:
The written-on-paper formula is probably capital `A`, but conflicts with Python's guidelines of reserving caps for constants. Maybe rename `A` to be  a more descriptive variable?